nosecone/index.ts

```diff
@@ -143,18 +143,168 @@ export interface PermittedCrossDomainPoliciesConfig {
+   * - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+   * - https://owasp.org/www-project-secure-headers/#strict-transport-security
+   */
   strictTransportSecurity?: StrictTransportSecurityConfig | boolean;
+  /**
+   * Configure the `X-Content-Type-Options` header, which helps mitigate MIME
+   * type sniffing that can cause security issues.
+   *
+   * Enable with defaults by specifying `true` or disable by specifying `false`.
+   *
+   * See also:
+   * - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+   * - https://developer.mozilla.org/en-US/docs/Web/HTTP/MIME_types#mime_sniffing
+   * - https://owasp.org/www-project-secure-headers/#x-content-type-options
+   */
   xContentTypeOptions?: boolean;
+  /**
+   * Configure the `X-DNS-Prefetch-Control` header, which helps control DNS
+   * prefetching to improve user privacy at the expense of performance.
+   *
+   * Enable with defaults by specifying `true`, disable by specifying `false`,
+   * or provide {@link DnsPrefetchControlConfig} to configure individual items.
+   *
+   * See also:
+   * - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-DNS-Prefetch-Control
+   */
   xDnsPrefetchControl?: DnsPrefetchControlConfig | boolean;
+  /**
+   * Configure the `X-Download-Options` header, which prevents a user from
+   * opening a file directly in Internet Explorer 8 to avoid prevent script
+   * injection.
+   *
+   * Enable with defaults by specifying `true` or disable by specifying `false`.
+   *
+   * See also:
+   * - https://learn.microsoft.com/en-us/archive/blogs/ie/ie8-security-part-v-comprehensive-protection#mime-handling-force-save
+   */
   xDownloadOptions?: boolean;
+  /**
+   * Configure the `X-Frame-Options` header, which help mitigate clickjacking
+   * attacks in legacy browsers. This header is superceded by a directive in the
+   * `Content-Security-Policy` header.
+   *
+   * Enable with defaults by specifying `true`, disable by specifying `false`,
+   * or provide {@link FrameOptionsConfig} to configure individual items.
+   *
+   * See also:
+   * - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
+   * - https://owasp.org/www-project-secure-headers/#x-frame-options
+   */
   xFrameOptions?: FrameOptionsConfig | boolean;
+  /**
+   * Configure the `X-Permitted-Cross-Domain-Policies` header, which tells some
+   * clients, like Adobe products, your domain's policy for loading cross-domain
+   * content.
+   *
+   * Enable with defaults by specifying `true`, disable by specifying `false`,
+   * or provide {@link PermittedCrossDomainPoliciesConfig} to configure
+   * individual items.
+   *
+   * See also:
+   * - https://owasp.org/www-project-secure-headers/#x-permitted-cross-domain-policies
+   */
   xPermittedCrossDomainPolicies?: PermittedCrossDomainPoliciesConfig | boolean;
+  /**
+   * Disable the `X-XSS-Protection` header, which could introduce a browser
```

Comment:
This is deprecated and doesn't work consistently across browsers. Should we add some note about that?

Reply:
Yeah fine if it's disabled by default.

Would love to see if any browsers are still using this! Must be years old at this point:

<img width="688" alt="image" src="https://github.com/user-attachments/assets/ac2603c4-a039-4df9-a42f-8c472efa7d9a">